test/benchmarks/src/main/scala/scala/collection/mutable/OpenHashMapBenchmark.scala

```diff
@@ -0,0 +1,204 @@
+  @State(Scope.Thread)
+  class BulkRemovedGetState {
+    val map = new OpenHashMap[Int,Int].empty
+
+    /** Load the map with keys from `1` to `size`, removing half of them. */
+    @Setup
+    def setup(params: BenchmarkParams) {
+      val size = params.getParam("size").toInt
+      put_remove_Int(map, size)
+      System.gc()
+    }
+  }
+
+  /** Put elements into the given map. */
+  private def put_Int(map: OpenHashMap[Int,Int], from: Int, to: Int) {
+    var i = from
+    while (i <= to) {  // using a `for` expression instead adds significant overhead
+      map.put(i, i)
+      i += 1
+    }
+  }
+
+  /** Put elements into the given map, removing half of them as they're added.
+    * 
+    * @param size number of entries to leave in the map on return
+    */
+  def put_remove_Int(map: OpenHashMap[Int,Int], size: Int) {
+    val blocks = 50  // should be a factor of `size`
+    val totalPuts = 2 * size  // add twice as many, because we remove half of them
+    val blockSize: Int = totalPuts / blocks
+    var base = 0
+    while (base < totalPuts) {
+      put_Int(map, base + 1, base + blockSize)
+
+      // remove every other entry
+      var i = base + 1
+      while (i <= base + blockSize) {
+        map.remove(i)
+        i += 2
+      }
+
+      base += blockSize
+    }
+  }
+
+  /** Get elements from the given map. */
+  def get_Int(map: OpenHashMap[Int,Int], size: Int, bh: Blackhole) {
+    var i = 1
+    while (i <= size) {
+      bh.consume(map.get(i).getOrElse(0))
+      i += 1
+    }
+  }
+}
+
+/** Benchmark for the library's [[OpenHashMap]]. */
+@BenchmarkMode(Array(Mode.AverageTime))
+@Fork(6)
+@Threads(1)
+@Warmup(iterations = 20)
+@Measurement(iterations = 6)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Benchmark)
+class OpenHashMapBenchmark {
+  import OpenHashMapBenchmark._
+
+  @Param(Array("25", "50", "100", "250", "1000", "2500", "10000", "25000", "100000", "250000", "1000000", "2500000",
+               "5000000", "7500000", "10000000", "25000000"))
+  var size: Int = _
+
+  /** Test putting elements to a map of `Int` to `Int`. */
+  @Benchmark
+  def put_Int(state: BulkPutState) {
+    var i = 0
+    while (i < state.maps.length) {
+      OpenHashMapBenchmark.put_Int(state.maps(i), 1, size)
+      i += 1
+    }
+  }
+
+  /** Test putting and removing elements to a growing map of `Int` to `Int`. */
+  @Benchmark
+  def put_remove_Int(state: BulkPutState) {
+    var i = 0
+    while (i < state.maps.length) {
+      OpenHashMapBenchmark.put_remove_Int(state.maps(i), size)
+      i += 1
+    }
+  }
+
+  /** Test getting elements from a map of `Int` to `Int`. */
+  @Benchmark
+  def put_get_Int(state: BulkGetState, bh: Blackhole) = OpenHashMapBenchmark.get_Int(state.map, size, bh)
+
+  /** Test getting elements from a map of `Int` to `Int` from which elements have been removed. */
+  @Benchmark
+  def put_remove_get_Int(state: BulkRemovedGetState, bh: Blackhole) = OpenHashMapBenchmark.get_Int(state.map, size, bh)
```

Comment:
I agree with not holding up the PR.  My thought was that my attention is very limited, but I did spend some of it here, so why not take care of the problems now (if you agree there are improvements to be made)?  (Possible reason: because someone else will be reviewing the other PRs, so you won't be held back by how busy I am.)